### PR TITLE
fix: duplicated files when "skipFilesWithNoCoverage=false" on windows

### PR DESCRIPTION
**Pull request by CrisDan1905**
When "skipFilesWithNoCoverage=false" on Windows, files are duplicated and the coverage percentage is wrong. It's because istanbul returns routes with mixed forward/backslashes in the paths. So a new utility function was added for removing forwardslashes instead of replace them, so when comparing istambul routes with the ones stored, they may be compared rightly, so a path like:

 `\\Users\\mattlewis\\Code\\/open-source\\/karma-coverage-istanbul-reporter\\test\\fixtures\\typescript`

is transformed into:
`\\Users\\mattlewis\\Code\\open-source\\karma-coverage-istanbul-reporter\\test\\fixtures\\typescript`

instead of:
`\\Users\\mattlewis\\Code\\\\open-source\\\\karma-coverage-istanbul-reporter\\test\\fixtures\\typescript`

### Listing the file:
image

### Inside the file:
image

### Listing the file:
image

### Inside the file:
image

### CODE REVIEW

**Comment by CrisDan1905** [comment]
I'm noticing that some style changes that break Travis-ci are applied when I push my changes. Not sure why it's happening

**Comment by CrisDan1905** [comment]
@mattlewis92 Sure!, `return filePath.replace(/\//g, '\\').replace(/\\\\/g, '\\'); ` works well too 🙂

**Comment by CrisDan1905** [comment]
If you prefer I can change the PR to only change  that line you mentioned

**Comment by mattlewis92** [comment]
> If you prefer I can change the PR to only change that line you mentioned

Yes please! As I think it'll need to be applied in all places that `fixPathSeparators` is called. 

Out of curiosity was this working before with version 2? I'm just trying to work out if the istanbul upgrade in version 3 of the lib was what broke it

**Comment by CrisDan1905** [comment]
> > If you prefer I can change the PR to only change that line you mentioned
> 
> Yes please! As I think it'll need to be applied in all places that `fixPathSeparators` is called.
> 
> Out of curiosity was this working before with version 2? I'm just trying to work out if the istanbul upgrade in version 3 of the lib was what broke it

Yes! reports were working perfectly fine until v2.0.5.

**Comment by CrisDan1905** [comment]
@mattlewis92 sorry, I mean, v.2.0.1

**Comment by mattlewis92** [comment]
Nice, thanks! Would you mind adding a test for your changes? You should be able to copy and adapt [this one](https://github.com/mattlewis92/karma-coverage-istanbul-reporter/blob/master/test/util.spec.js#L58-L80)

**Comment by CrisDan1905** [comment]
Alright, I added UT for the util function, let me know if it's ok to you or any changes you suggest :)

File: test/util.spec.js [hunk 1]
```diff
@@ -244,4 +245,25 @@ describe('util', () => {
...
   });
+
+  describe('fixPathSeparators', () => {
+    it('Should transform any forwardslash into backslash and remove any leftover backslash if OS is Windows', () => {
+      let output;
+
+      const input =
+        '\\Users\\mattlewis\\Code\\/open-source/karma-coverage-istanbul-reporter\\test\\fixtures\\typescript';
+
+      const isWin = process.platform.startsWith('win');
```
Comment (mattlewis92) [inline test/util.spec.js:256]: Can you change this to:
```
Object.defineProperty(process, 'platform', {
        value: 'win32',
      });
```

and then you can remove the `isWin` check below so the test will always run regardless of if you're on windows or not. Other than that, LGTM! 😄

Comment (CrisDan1905) [reply test/util.spec.js]: Done, I divided this Unit test into two, one for Windows and other for Linux, thanks!

**Pull request merged**